cgosqlite/cgosqlite.go

```diff
@@ -118,6 +118,19 @@ func (db *DB) BusyTimeout(d time.Duration) {
 	C.sqlite3_busy_timeout(db.db, C.int(d/1e6))
 }
 
+func (db *DB) Checkpoint(dbName string, mode sqliteh.Checkpoint) (int, int, error) {
+	var cDB *C.char
+	if dbName != "" {
+		// Docs say: "If parameter zDb is NULL or points to a zero length string",
+		// so they are equivalent here.
+		cDB = C.CString(dbName)
+		defer C.free(unsafe.Pointer(cDB))
```

Comment:
`defer` in an if statement (instead of top level) is more expensive. Seems like checkpointing is expensive enough that that doesn't matter, but thought I should mention it just in case.